ddln_imaginator.proto

```diff
@@ -36,111 +52,90 @@ message Environment {
     }
-    repeated Clouds clouds = 1;
+    repeated Clouds clouds = 2; // Up to 3 layers supported
 
-    // dry particles in the air. it keeps the brightness and color of the light.
+    // Haze: simulates dry particles in the air. Acts like a fog but keeps the brightness and color of the light.
     message Haze {
         enum Type {
-            // default keeps the value; visibility = 20000, density = 1
-            DEFAULT = 0;
-
-            // Custom visibility and density as specified below
-            CUSTOM = 1;
+            DEFAULT = 0; // default setting, e.g. visibility = 20000, density = 1
+            CUSTOM = 1; // Custom visibility and density as specified below
         }
         optional Type type = 1;
-
-        // visibility distance: 0 - 20000
-        optional float visibility = 2;
-
-        // density value: 0.001 - 100
-        optional float density = 3;
+        optional float visibility = 2;  // visibility distance [m]
+        optional float density = 3; // density value:, typical range 0.001..100 [-]
     }
-    optional Haze haze = 2;
+    optional Haze haze = 3;
 
-    // water droplets in the air. white colored fog which obstructs the light source.
+    // Fog: water droplets in the air. White colored fog which obstructs the light source.
     message Fog {
         optional bool enabled = 1;
-
-        // fog multiplier value: 0.01 - 0.0001
-        optional float multiplier = 2 ;
+        optional float multiplier = 2 ;  // fog multiplier value, typical range 0.01..0.0001 [-]
     }
-    optional Fog fog = 3;
+    optional Fog fog = 4;
 
+    // Control sun position and intensity
     message Sun {
         enum Type {
-            // default uses the sun attached with the world. parameters are not changeable.
-            DEFAULT = 0;
-
-            // customized sun with parameters below
-            CUSTOM = 1;
+            DEFAULT = 0; // Default uses the sun attached with the world. parameters are not changeable.
+            CUSTOM = 1; // customized sun with parameters below
         }
         optional Type type = 1;
-        optional float azimuth_deg = 2;
-
-        // altitude should not exceed 90 degrees.
-        optional float altitude_deg = 3;
-
-        // intensity value: 0 - 1
-        optional float intensity = 4;
+        optional float azimuth_deg = 2; // Azimuth sun angle, range 0..360 [deg]
+        optional float altitude_deg = 3; // altitude, ranges 0..90 [deg]
+        optional float intensity = 4; // intensity value, typical range 0..1 [-]
     }
-    optional Sun sun = 4;
+    optional Sun sun = 5;
 
+    // Configure rain particles.
+    // Note 1: The particles are rendered in real time, e.g. subsequent image requests should be made in real time
+    // for realistic rendering.
+    // Note 2: Rain sprites always appear to fall down vertically since camera velocity is not taken into account.
```

Comment:
details of the current implementation should be kept out of the interface definition file
here we specify what is the intent of the message

Reply:
This message is indicating an limitation. The user should be informed IMO.